simulator/src/main/java/com/hedera/block/simulator/generator/GeneratorInjectionModule.java

```diff
@@ -16,22 +16,34 @@
 
 package com.hedera.block.simulator.generator;
 
-import dagger.Binds;
+import com.hedera.block.simulator.config.data.BlockStreamConfig;
 import dagger.Module;
+import dagger.Provides;
 import javax.inject.Singleton;
 
 /** The module used to inject the block stream manager. */
 @Module
 public interface GeneratorInjectionModule {
 
     /**
-     * Provides the block stream manager.
+     * Provides the block stream manager based on the configuration, either
+     * BlockAsFileBlockStreamManager or BlockAsDirBlockStreamManager. by default, it will be
+     * BlockAsFileBlockStreamManager.
      *
-     * @param blockAsFileBlockStreamManager the block as file block stream manager
+     * @param config the block stream configuration
      * @return the block stream manager
      */
     @Singleton
-    @Binds
-    BlockStreamManager provideBlockStreamManager(
-            BlockAsFileBlockStreamManager blockAsFileBlockStreamManager);
+    @Provides
+    static BlockStreamManager providesBlockStreamManager(BlockStreamConfig config) {
+
+        if ("BlockAsFileBlockStreamManager".equalsIgnoreCase(config.managerImplementation())) {
+            return new BlockAsFileBlockStreamManager(config);
+        } else if ("BlockAsDirBlockStreamManager"
+                .equalsIgnoreCase(config.managerImplementation())) {
+            return new BlockAsDirBlockStreamManager(config);
+        }
+
+        return new BlockAsFileBlockStreamManager(config);
```

Comment:
I think you can refactor this method to something like:
```
if ("BlockAsDirBlockStreamManager"
                .equalsIgnoreCase(config.managerImplementation())) {
            return new BlockAsDirBlockStreamManager(config);
}

return new BlockAsFileBlockStreamManager(config);
```

Reply:
Agree with Matt, and later if there other cases, we can do enum with switch, for example.

Reply:
Great catch, Thanks!


✅  Done!